transaction.go

```diff
@@ -569,31 +587,61 @@ type ProposalKey struct {
 
 // A TransactionSignature is a signature associated with a specific account key.
 type TransactionSignature struct {
-	Address     Address
-	SignerIndex int
-	KeyIndex    uint32
-	Signature   []byte
+	Address       Address
+	SignerIndex   int
+	KeyIndex      uint32
+	Signature     []byte
+	ExtensionData []byte
 }
 
 type transactionSignatureCanonicalForm struct {
-	SignerIndex uint
-	KeyIndex    uint32
-	Signature   []byte
-}
-
-func (s TransactionSignature) canonicalForm() transactionSignatureCanonicalForm {
+	SignerIndex   uint
+	KeyIndex      uint32
+	Signature     []byte
+	ExtensionData []byte
+}
+
+var _ transactionSignatureCommonForm = (*transactionSignatureCanonicalForm)(nil)
+
+func (s transactionSignatureCanonicalForm) dummy() {}
+
+// Checks if the scheme is plain authentication scheme, and indicate that it
+// is required to use the legacy canonical form.
+// We check for a valid scheme identifier, as this should be the only case
+// where the extension data can be left out of the cannonical form.
+// All other non-valid cases that are similar to the plain scheme, but is not valid,
+// should be included in the canonical form, as they are not valid signatures
+func (s TransactionSignature) shouldUseLegacyCanonicalForm() bool {
+	plainSchemeIdentifier := byte(0)
```

Comment:
This seems to be a constant for a magic number
```suggestion
	const plainSchemeIdentifier byte = 0
```